substrate/frame/transaction-payment/asset-tx-payment/src/benchmarking.rs

```diff
@@ -0,0 +1,135 @@
+
+use super::*;
+use crate::Pallet;
+use frame_benchmarking::v2::*;
+use frame_support::{
+	dispatch::{DispatchInfo, PostDispatchInfo},
+	pallet_prelude::*,
+	traits::fungibles::Inspect,
+};
+use frame_system::RawOrigin;
+use sp_runtime::traits::{AsSystemOriginSigner, DispatchTransaction, Dispatchable};
+
+pub trait ExtConfig: Config {
+	fn create_asset_id_parameter(
+		id: u32,
+	) -> (
+		<<Self as Config>::Fungibles as Inspect<Self::AccountId>>::AssetId,
+		<<Self as Config>::OnChargeAssetTransaction as OnChargeAssetTransaction<Self>>::AssetId,
+	);
+	fn setup_balances_and_pool(
+		asset_id: <<Self as Config>::Fungibles as Inspect<Self::AccountId>>::AssetId,
+		account: Self::AccountId,
+	);
+}
+
+#[benchmarks(where
+	T: Config + Send + Sync + ExtConfig,
+	T::RuntimeCall: Dispatchable<Info = DispatchInfo, PostInfo = PostDispatchInfo>,
+	AssetBalanceOf<T>: Send + Sync,
+	BalanceOf<T>: Send + Sync + From<u64> + IsType<ChargeAssetBalanceOf<T>>,
+	ChargeAssetIdOf<T>: Send + Sync,
+	<T::RuntimeCall as Dispatchable>::RuntimeOrigin: AsSystemOriginSigner<T::AccountId> + Clone,
+    Credit<T::AccountId, T::Fungibles>: IsType<ChargeAssetLiquidityOf<T>>,
+)]
+mod benchmarks {
+	use super::*;
+
+	#[benchmark]
+	fn charge_asset_tx_payment_zero() {
+		let caller: T::AccountId = whitelisted_caller();
+		let ext: ChargeAssetTxPayment<T> = ChargeAssetTxPayment::from(0u32.into(), None);
+		let inner = frame_system::Call::remark { remark: vec![] };
+		let call = T::RuntimeCall::from(inner);
+		let info = DispatchInfo {
+			weight: Weight::zero(),
+			class: DispatchClass::Normal,
+			pays_fee: Pays::No,
+		};
+		let post_info = PostDispatchInfo { actual_weight: None, pays_fee: Pays::No };
+		#[block]
+		{
+			assert!(ext
+				.test_run(RawOrigin::Signed(caller).into(), &call, &info, 0, |_| Ok(post_info))
+				.unwrap()
+				.is_ok());
+		}
+	}
+
+	#[benchmark]
+	fn charge_asset_tx_payment_native() {
+		let caller: T::AccountId = whitelisted_caller();
+		let (fun_asset_id, _) = T::create_asset_id_parameter(1);
+		T::setup_balances_and_pool(fun_asset_id, caller.clone());
+		let ext: ChargeAssetTxPayment<T> = ChargeAssetTxPayment::from(10u32.into(), None);
+		let inner = frame_system::Call::remark { remark: vec![] };
+		let call = T::RuntimeCall::from(inner);
+		let info = DispatchInfo {
+			weight: Weight::from_parts(10, 0),
+			class: DispatchClass::Operational,
+			pays_fee: Pays::Yes,
+		};
+		let post_info = PostDispatchInfo {
+			actual_weight: Some(Weight::from_parts(10, 0)),
+			pays_fee: Pays::Yes,
+		};
+
+		#[block]
+		{
+			assert!(ext
+				.test_run(RawOrigin::Signed(caller).into(), &call, &info, 0, |_| Ok(post_info))
+				.unwrap()
+				.is_ok());
+		}
+	}
+
+	#[benchmark]
+	fn charge_asset_tx_payment_asset() {
+		let caller: T::AccountId = whitelisted_caller();
+		let (fun_asset_id, asset_id) = T::create_asset_id_parameter(1);
+		T::setup_balances_and_pool(fun_asset_id.clone(), caller.clone());
+		let tip = 10u32.into();
```

Comment:
Perhaps add the similar `minimum_balance` here?

Reply:
I think it's fine with the hardcoded `10` tip either way, all benches run fine. Depending on how quickly this PR moves, I might have time to make this change too since I would need to re-run all the benches for this pallet to make sure they work with the change.